test/config.test.js

```diff
@@ -1596,3 +1596,21 @@ test('spanStackTraceMinDuration', suite => {
 
   suite.end()
 })
+
+test('env variable names', suite => {
+  // flatten
+  const names = [].concat(...Object.values(config.ENV_TABLE))
+
+  // list of names we keep around for backwards compatability
+  // but that don't conform to the ELASTIC_APM name
+  const legacy = ['ELASTIC_SANITIZE_FIELD_NAMES', 'KUBERNETES_POD_UID',
+    'KUBERNETES_POD_NAME', 'KUBERNETES_NODE_NAME', 'KUBERNETES_NAMESPACE',
+    'ELASTIC_IGNORE_MESSAGE_QUEUES']
+  for (const name of names) {
+    if (legacy.indexOf(name) !== -1) {
```

Comment:
The `KUBERNETES_*` envvars are actually the correct ones (according to https://github.com/elastic/apm-agent-nodejs/issues/1028), so it is misleading to mark them as legacy here.

```suggestion
  // List of envvar names that are allowed to not have the `ELASTIC_APM_`
  // prefix.
  const allowedNoPrefix = [
    // Kubernetes config vars intentionally do not use the prefix.
    // https://github.com/elastic/apm/issues/21
    'KUBERNETES_POD_UID',
    'KUBERNETES_POD_NAME',
    'KUBERNETES_NODE_NAME',
    'KUBERNETES_NAMESPACE',
    // Backwards compat: https://github.com/elastic/apm-agent-nodejs/pull/2637
    'ELASTIC_SANITIZE_FIELD_NAMES', 
    'ELASTIC_IGNORE_MESSAGE_QUEUES'
  ]
  for (const name of names) {
    if (allowedNoPrefix.indexOf(name) !== -1) {
```